LongestSubString.java

```diff
@@ -0,0 +1,108 @@
+/*
+Passed All Test Cases 
+
+HashSet Approach 
+
+Big O 
+Time-> O(N), N is # of chars in string s
+Space-> O(N), N is # of chars in string s

```

Comment:
Space here is O(1) as well. We are just using a hashset. In worst case it will be taking 256 entries. Worst case scenario of time will never be dependent on  N that is length of string